.github/workflows/release.yml

```diff
@@ -78,16 +81,16 @@ jobs:
         with:
           python-version: ${{ matrix.python_version }}
           architecture: ${{ matrix.arch.python }}
-      - run: python -m pip install -U pip wheel setuptools pytest pyopenssl
       - uses: msys2/setup-msys2@v2
         with:
           msystem: MINGW${{ matrix.arch.mingw }}
           update: true
+          release: false
```

Comment:
What does this do?

Reply:
https://github.com/msys2/setup-msys2#release

It instructs the github action to use the already installed msys version and just update the packages, but it supports caching. It is almost the same as without, the only difference is:
 1. with `false` msys can be reached at `C:\msys64` instead of a temporary folder, and this makes extending the PATH for building from python easier.
 2. There might be some extra packages, but they shouldn't cause any issues.